src/main/java/com/greenfox/poker/service/UserService.java

```diff
@@ -2,77 +2,114 @@
 
 import com.greenfox.poker.model.LoginRequest;
 import com.greenfox.poker.model.PokerUser;
+import com.greenfox.poker.model.PokerUserDTO;
+import com.greenfox.poker.model.ResponseType;
 import com.greenfox.poker.model.UserTokenResponse;
 import com.greenfox.poker.model.StatusError;
+import com.greenfox.poker.repository.PokerUserRepo;
 import java.util.ArrayList;
 import java.util.List;
-import org.springframework.http.HttpStatus;
-import org.springframework.http.ResponseEntity;
+import java.util.stream.Collectors;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
 import org.springframework.validation.BindingResult;
 import org.springframework.validation.FieldError;
 
 @Component
 public class UserService {
 
-  public ResponseEntity<?> mockRespondToSuccessfulRegisterOrLogin() {
-    return new ResponseEntity(new UserTokenResponse("success", "ABC123", 4321), HttpStatus.OK);
+  @Autowired
+  PokerUserRepo pokerUserRepo;
+
+  @Autowired
+  PokerUserDTO pokerUserDTO;
+
+  @Autowired
+  TokenService tokenService;
+
+  public ResponseType responseToSuccessfulRegister(PokerUser pokerUser) {
+    pokerUserRepo.save(pokerUser);
+    String token = tokenService.generateToken(pokerUser);
+    return new UserTokenResponse("success", token, pokerUser.getId());
   }
 
-  public ResponseEntity<?> respondToMissingParameters(BindingResult bindingResult) {
-    List<String> listOfMissingFields = new ArrayList<>();
+  public ResponseType responseToSuccessfulLogin(LoginRequest loginRequest) {
+    PokerUser pokerUserFromDatabase = pokerUserRepo.findByUsername(loginRequest.getUsername()).get(0);
+    String token = tokenService.generateToken(pokerUserFromDatabase);
+    return new UserTokenResponse("success", token, pokerUserFromDatabase.getId());
+  }
+
+  public ResponseType respondToMissingParameters(BindingResult bindingResult) {
```

Comment:
That method is OK right now, but don't be surprised if we'll refactor this later.